Xamarin.Forms.Controls.Issues/Xamarin.Forms.Controls.Issues.Shared/Github5623.xaml.cs

```diff
@@ -0,0 +1,168 @@
+﻿using System.Collections.Generic;
+using System.Collections.ObjectModel;
+using System.Threading;
+using System.Threading.Tasks;
+using Xamarin.Forms.CustomAttributes;
+using Xamarin.Forms.Internals;
+using Xamarin.Forms.Xaml;
+
+#if UITEST
+using Xamarin.UITest;
+using NUnit.Framework;
+using Xamarin.Forms.Core.UITests;
+using System.Linq;
+#endif
+
+namespace Xamarin.Forms.Controls.Issues
+{
+#if UITEST
+	[Category(UITestCategories.CollectionView)]
+#endif
+#if APP
+	[XamlCompilation(XamlCompilationOptions.Compile)]
+#endif
+	[Preserve(AllMembers = true)]
+	[Issue(IssueTracker.Github, 5623, "CollectionView with Incremental Collection (RemainingItemsThreshold)", PlatformAffected.All)]
+	public partial class Github5623 : TestContentPage
+	{
+		int _itemCount = 100;
+		int _maximumItemCount = 1000;
+		int _pageSize = 100;
+		static SemaphoreSlim SemaphoreSlim = new SemaphoreSlim(1, 1);
+
+		public Github5623()
+		{
+#if APP
+			Device.SetFlags(new List<string> { CollectionView.CollectionViewExperimental });
+
+			InitializeComponent();
+
+			BindingContext = new ViewModel5623();
+#endif
+		}
+
+		protected override void Init()
+		{
+
+		}
+
+		private async void CollectionView_RemainingItemsThresholdReached(object sender, System.EventArgs e)
+		{
+			await SemaphoreSlim.WaitAsync();
```

Comment:
The event/command will be called multiple times if the user scrolls down like a mad cow (which also depends on cell size, web service latency, and handler processing time). We can serialize event handlers and use a control variable (such as continuation token being null after the last web service call) to stop processing unwanted invocations.